apps/db_assembler/includes/functions.sh

```diff
@@ -7,42 +7,49 @@ function dbasm_resetExitCode() {
 	exit 0
 }
 
+function dbasm_resetMYSQLPort()
+{
+	MYSQL_PORT=3306
```

Comment:
Can you explain me why do we need to reset the port everytime? the default value is already inside the config.sh.dist file that is loaded first

Reply:
So, this was added for backwards compatibility, where users still using the old config files and only added the 'MYSQL_PORT' to one of the db config, thus resulting in the following one using the wrong port (the custom one instead of the default one).
For example:
```
DB_AUTH_CONF="MYSQL_USER='root'; \
                    MYSQL_PASS=''; \
                    MYSQL_HOST='127.0.0.1';\
					MYSQL_PORT='3336';\
                    "

DB_CHARACTERS_CONF="MYSQL_USER='root'; \
                    MYSQL_PASS=''; \
                    MYSQL_HOST='127.0.0.1';\
                    "

```
In this example without resseting the port, the db will use the port 3336, as it was the last one set, so reseting this would mitigate this sort of situation of user error.

Reply:
Ok clear. This is an edge case since the DB_*_CONF are evaluated in those methods. However, I'm sorry to say something "against" @BarbzYHOOL but I think for DX (developer experience) is better a one-line native syntax for default values.

Check parameter-substitution documentation: https://tldp.org/LDP/abs/html/parameter-substitution.html

Here you can find a proper example: https://tldp.org/LDP/abs/html/parameter-substitution.html

Something like: ${MYSQL_PORT:=3306}

Reply:
it's a non common syntax so a comment would be appreciated:

```
# if MYSQL_PORT not set, use default port
${MYSQL_PORT:=3306}
```

Reply:
Thanks for the links to documentation!
I have been playing around with them for a while, but they cannot be used to resolve the current edge case, at least in my bash (wls ubuntu) , the value for MYSQL_PORT is carried over, so MYSQL_PORT is initialized to whatever value was used in the last DB operation in my test case to 3336. 
Example Config:
```
DB_AUTH_CONF="MYSQL_USER='root'; \
                    MYSQL_PASS=''; \
                    MYSQL_HOST='127.0.0.1';\
                    "

DB_CHARACTERS_CONF="MYSQL_USER='root'; \
                    MYSQL_PASS=''; \
                    MYSQL_HOST='127.0.0.1';\
					MYSQL_PORT='3336';\
                    "

DB_WORLD_CONF="MYSQL_USER='root'; \
                    MYSQL_PASS=''; \
                    MYSQL_HOST='127.0.0.1';\
                    "
DB_AUTH_NAME="acore_auth1"
DB_CHARACTERS_NAME="acore_characters"
DB_WORLD_NAME="acore_world"
```

Output (The output is done just before the eval):
```
3306
acore_auth1 is not empty, base importing skipped
3306
acore_characters is not empty, base importing skipped
3336
acore_world is not empty, base importing skipped`

```
As you can see acore_world is initialized to 3336 and the ${MYSQL_PORT:=3306} will not get the behavior we want; So the solution is either setting it it 3306 every time.
We could use parameter substitution but it would be 2 calls (${MYSQL_PORT:=3306} and ${MYSQL_PORT+3306}) but it is more clear to use the an asigment imho.

The solution would look like this just before each eval:
```
# MYSQL_PORT needs to be reseted as the next eval might not overwite the current value causing the commands to use wrong port
MYSQL_PORT=3306
```

Reply:
Looks good for me :) I just don't see the need of a dedicated function for just a single value